### DISCUSSION TITLE
Texture from html canvas for SimpleMeshLayer

### DISCUSSION THREAD

**Discussion by bolatuly** [visgl/deck.gl#7765] [Q&A] [answered]
Hi,

I'm trying to use HTML canvas as a texture for SimpleMeshLayer. But seems like canvas is used only in the initiation of texture. My goal is to update the canvas after some time, so that texture of the mesh updates too. Can you please suggest a way to achieve this?

I've created a codepen sample to show what I'm trying to achieve. https://codesandbox.io/s/deck-gl-texture-canvas-nldy5c

**Comment by Pessimistress** [comment, marked as answer]
WebGL texture only reads your canvas pixels once upon instantiation. You need to tell the texture that the source has changed.

If your canvas does not redraw frequently, you can just create a new texture object upon rerender (see [docs](https://deck.gl/docs/api-reference/mesh-layers/simple-mesh-layer#texture)):

```js
new SimpleMeshLayer({
  texture: {data: canvas, width: 100, height: 100}  // shallow change will recreate the texture
})
```

If you are going to redraw it on animation frame or something, you can also manually update the `Texture2D` instance:

```js
// Update 2d canvas
mesh_layer.props.texture?.initialize?.({data: canvas});
mesh_layer.setNeedsRedraw();
```

**Reply by bolatuly** [reply]
The final goal is to play RTSP stream from IP camera into html canvas. So I need to redraw mesh layer's texture on each frame of RTSP stream.
